cli/ops.rs

```diff
@@ -2194,3 +2201,135 @@ fn op_get_random_values(
 }
+
+fn op_dynamic_lib_load(
+  state: &ThreadSafeState,
+  base: &msg::Base<'_>,
+  _data: Option<PinnedBuf>,
+) -> Box<OpWithError> {
+  let cmd_id = base.cmd_id();
+  let inner = base.inner_as_dynamic_lib_load().unwrap();
+  let (filename, filename_) = match resolve_path(inner.filename().unwrap()) {
+    Err(err) => return odd_future(err),
+    Ok(v) => v,
+  };
+
+  if let Err(e) = state.check_dlopen(&filename_) {
+    return odd_future(e);
+  }
+
+  Box::new(futures::future::result(move || -> OpResult {
+    let lib = bindings::load_binding(filename)?;
+    let lib_id: bindings::DylibId =
+      state.next_dylib_id.fetch_add(1, Ordering::SeqCst);
+
+    let mut loaded_libs = state.loaded_dylibs.write().unwrap();
+    loaded_libs.insert(lib_id, lib);
+
+    let builder = &mut FlatBufferBuilder::new();
+    let msg_inner = msg::DynamicLibLoadRes::create(
+      builder,
+      &msg::DynamicLibLoadResArgs { lib_id },
+    );
+
+    Ok(serialize_response(
+      cmd_id,
+      builder,
+      msg::BaseArgs {
+        inner: Some(msg_inner.as_union_value()),
+        inner_type: msg::Any::DynamicLibLoadRes,
+        ..Default::default()
+      },
+    ))
+  }()))
+}
+
+fn op_dynamic_lib_fn_load(
+  state: &ThreadSafeState,
+  base: &msg::Base<'_>,
+  _data: Option<PinnedBuf>,
+) -> Box<OpWithError> {
+  let cmd_id = base.cmd_id();
+  let inner = base.inner_as_dynamic_lib_fn_load().unwrap();
+  let lib_id = inner.lib_id();
+  let name = inner.name().unwrap();
+
+  Box::new(futures::future::result(move || -> OpResult {
+    let loaded_libs = state.loaded_dylibs.read().unwrap();
+    let lib = match loaded_libs.get(&lib_id) {
+      Some(lib) => lib,
+      None => return Err(errors::binding_bad_lib_id()),
+    };
+
+    let fun: OpDispatchFn =
+      *unsafe { lib.symbol::<OpDispatchFn>(name) }.unwrap();
+    let fn_id: bindings::DylibId =
+      state.next_dylib_fn_id.fetch_add(1, Ordering::SeqCst);
+    let mut loaded_fns = state.loaded_dylib_functions.write().unwrap();
+    loaded_fns.insert(fn_id, fun);
+
+    let builder = &mut FlatBufferBuilder::new();
+    let msg_inner = msg::DynamicLibFnLoadRes::create(
+      builder,
+      &msg::DynamicLibFnLoadResArgs { fn_id },
+    );
+
+    Ok(serialize_response(
+      cmd_id,
+      builder,
+      msg::BaseArgs {
+        inner: Some(msg_inner.as_union_value()),
+        inner_type: msg::Any::DynamicLibFnLoadRes,
+        ..Default::default()
+      },
+    ))
+  }()))
+}
+
+fn op_dynamic_lib_fn_call(
+  state: &ThreadSafeState,
+  base: &msg::Base<'_>,
+  data: Option<PinnedBuf>,
+) -> Box<OpWithError> {
+  let cmd_id = base.cmd_id();
+  let inner = base.inner_as_dynamic_lib_fn_call().unwrap();
+  let fn_id = inner.fn_id();
+
+  let loaded_fns = state.loaded_dylib_functions.read().unwrap();
+  let fun = match loaded_fns.get(&fn_id) {
+    Some(fun) => fun,
+    None => return odd_future(errors::binding_bad_fn_id()),
+  };
+
+  let result = fun(base.sync(), inner.data().unwrap(), data);
```

Comment:
Why do you need both `inner.data()` and `data` .... ? I would prefer if `fun` only took the zero-copy data.

Reply:
My primary rationale here is it matches the pattern of existing ops in deno.